### PR TITLE
Redo check_expire.py

### PR DESCRIPTION
**Pull request by hasegeli**

### CODE REVIEW

File: src/check_ssl_cert_expire.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+
+def get_extension_value(cert, short_name):
+    for ext_num in range(cert.get_extension_count()):
+        if cert.get_extension(ext_num).get_short_name() == short_name:
+            return str(cert.get_extension(ext_num))
+
+
+def decode_san(san_string):
+    if san_string:
```
Comment (TheFRedFox) [inline src/check_ssl_cert_expire.py]: I am not sure about yielding methods. But can you also do an early return check here?
``` python
if not san_string:
    return None  # or just return
```

File: src/check_ssl_cert_expire.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+from datetime import datetime, timedelta
+from dateutil.tz import tzutc
+from dateutil.parser import parse
+from sys import exit
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        'pemfile',
```
Comment (TheFRedFox) [inline src/check_ssl_cert_expire.py]: Do you already know about `argparse.FileType`? https://docs.python.org/2/library/argparse.html#filetype-objects

File: src/check_ssl_cert_expire.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        print('OK: ' + output)
+    elif exit_code == 1:
+        print('WARNING: ' + output)
+    elif exit_code == 2:
+        print('CRITICAL: ' + output)
+    exit(exit_code)
+
+
+def interval(arg):
+    index = next(
```
Comment (TheFRedFox) [inline src/check_ssl_cert_expire.py]: Didn't really read it completely, but it somehow looks overcomplicated. I am wondering if there is no other way to get the different values.

File: src/check_ssl_cert_expire.py (hunk 1 above)
Comment (hasegeli) [reply src/check_ssl_cert_expire.py]: Done

File: src/check_ssl_cert_expire.py (hunk 2 above)
Comment (hasegeli) [reply src/check_ssl_cert_expire.py]: Done.  Good to learn.

File: src/check_ssl_cert_expire.py (hunk 3 above)
Comment (hasegeli) [reply src/check_ssl_cert_expire.py]: Maybe it is good a have a little tricky code to not bore the reader.

Comment (TheFRedFox) [reply src/check_ssl_cert_expire.py]: 😆 Sometimes a good thing, yeah. But good then to have a TLDR comment above then. \^\^

File: src/check_ssl_cert_expire.py (hunk 2 above)
Comment (TheFRedFox) [reply src/check_ssl_cert_expire.py]: 👍

**Pull request merged**